src/fides/api/custom_types.py

```diff
@@ -70,89 +66,71 @@ class HtmlStr(str):
+    pattern = regex(r"^\+[1-9]\d{1,14}$")
+    if len(value) > max_length or len(value) < min_length or not pattern.search(value):
+        raise ValueError(
+            "Phone number must be formatted in E.164 format, i.e. '+15558675309'."
+        )
+    return value
+
 
-    @classmethod
-    def __get_validators__(cls) -> Generator:
-        yield cls.validate
-
-    @classmethod
-    def validate(cls, value: str) -> str:
-        # The front-end sends an empty string if the user doesn't input anything
-        if value == "":
-            return ""
-        max_length = 16  # Includes the +
-        min_length = 9
-        pattern = regex(r"^\+[1-9]\d{1,14}$")
-        if (
-            len(value) > max_length
-            or len(value) < min_length
-            or not pattern.search(value)
-        ):
-            raise ValueError(
-                "Phone number must be formatted in E.164 format, i.e. '+15558675309'."
-            )
-        return value
-
-
-class GPPMechanismConsentValue(str):
+PhoneNumber = Annotated[str, BeforeValidator(validate_phone_number)]
+
+
+def validate_gpp_mechanism_consent_value(value: str) -> str:
     """
-    Allowable consent values for GPP Mechanism Mappings.
+    Validator for GPPMechanismConsentValue -  Allowable consent values for GPP Mechanism Mappings.
     """
+    pattern = regex(r"^\d+$")
+    if not isinstance(value, str) or not pattern.search(value):
+        raise ValueError("GPP Mechanism consent value must be a string of digits.")
+    return value
 
-    @classmethod
-    def __get_validators__(cls) -> Generator:
-        yield cls.validate
 
-    @classmethod
-    def validate(cls, value: str) -> str:
-        pattern = regex(r"^\d+$")
-        if not isinstance(value, str) or not pattern.search(value):
-            raise ValueError("GPP Mechanism consent value must be a string of digits.")
-        return value
+GPPMechanismConsentValue = Annotated[
+    str, BeforeValidator(validate_gpp_mechanism_consent_value)
+]
 
 
-class URLOrigin(AnyUrl):
+def validate_path_of_url(value: AnyUrl) -> str:
     """
     A URL origin value. See https://developer.mozilla.org/en-US/docs/Glossary/Origin.
 
     We perform the basic URL validation, but also prevent URLs with paths,
     as paths are not part of an origin.
     """
+    # AnyURL now adds a trailing slash which will be considered a path, so stripping this off
+    if value.path and value.path != "/":
+        raise ValueError("URL origin values cannot contain a path.")
 
-    @classmethod
-    def __get_validators__(cls) -> Generator:
-        yield cls.validate
+    return str(value).rstrip("/")
 
-    @classmethod
-    def validate(cls, value: Any, field: ModelField, config: BaseConfig) -> AnyUrl:
-        value = super().validate(value, field, config)
-        if value.path:
-            raise ValueError("URL origin values cannot contain a path.")
 
-        return value
+URLOrigin = Annotated[AnyUrl, AfterValidator(validate_path_of_url)]
```

Comment:
AnyURL's add a trailing slash to everything now, and are used for cors_origins, we can't have trailing slashes there.  Updated the validator to strip it off.